images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh

```diff
@@ -933,16 +941,71 @@ if [[ "${CAPABILITIES[@]}" =~ "backup.appuio.ch/v1alpha1/Schedule" ]]; then
   fi
   TEMPLATE_PARAMETERS+=(-p BAAS_BUCKET_NAME="${BAAS_BUCKET_NAME}")
 
-  # Run Backups every day at 2200-0200
-  BACKUP_SCHEDULE=$( /kubectl-build-deploy/scripts/convert-crontab.sh "${NAMESPACE}" "M H(22-2) * * *")
+  # Pull in .lagoon.yml variables
+  LAGOON_YAML_KEYS=($(cat .lagoon.yml | shyaml keys))
+  if containsElement "production-backup-retention" "${LAGOON_YAML_KEYS[@]}"; then
+    LAGOON_YAML_BACKUP_RETENTION_KEYS=($(cat .lagoon.yml | shyaml keys production-backup-retention))
+
+    if containsElement "monthly" "${LAGOON_BACKUP_RETENTION_YAML_KEYS[@]}"; then
+      PRODUCTION_MONTHLY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml keys production-backup-retention.monthly -q)
+    else
+      PRODUCTION_MONTHLY_BACKUP_RETENTION=""
+    fi
+
+    if containsElement "weekly" "${LAGOON_BACKUP_RETENTION_YAML_KEYS[@]}"; then
+      PRODUCTION_WEEKLY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml keys production-backup-retention.weekly -q)
+    else
+      PRODUCTION_WEEKLY_BACKUP_RETENTION=""
+    fi
+
+    if containsElement "daily" "${LAGOON_BACKUP_RETENTION_YAML_KEYS[@]}"; then
+      PRODUCTION_DAILY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml keys production-backup-retention.daily -q)
+    else
+      PRODUCTION_DAILY_BACKUP_RETENTION=""
+    fi
+  else
+    PRODUCTION_MONTHLY_BACKUP_RETENTION=""
+    PRODUCTION_WEEKLY_BACKUP_RETENTION=""
+    PRODUCTION_DAILY_BACKUP_RETENTION=""
+  fi
```

Comment:
shyaml already has capability to handle non existing keys, see https://github.com/amazeeio/lagoon/blob/36cc7bda3a5b09fc63d19e2841cda8ebcd523ca6/images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh#L652-L658

so  you can just do:

```
    PRODUCTION_MONTHLY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml get-value production-backup-retention.monthly "")
    PRODUCTION_WEEKLY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml get-value production-backup-retention.weekly "")
    PRODUCTION_DAILY_BACKUP_RETENTION=$(cat .lagoon.yml | shyaml get-value production-backup-retention.daily "")
```

also there is `LAGOON_YAML_BACKUP_RETENTION_KEYS` created, but never used?

Reply:
I had originally built this because the call to `shyaml` was returning an error code if the specified value didn't exist, but I can no longer reproduce that error. I'll update the code accordingly.

About `LAGOON_YAML_BACKUP_RETENTION_KEYS`; this was the variable I assigned the array of keys in the `production-backup-retention` yaml structure to. I was using it on lines 949, 955, and 961 to set variables based on the existence of these keys. With the requested re-write here, I should be able to eliminate this altogether though.